receiver/mongodbatlasreceiver/alerts.go

```diff
@@ -0,0 +1,316 @@
+	if payloadSigHeader == "" {
+		rw.WriteHeader(http.StatusBadRequest)
+		a.logger.Debug("Got payload with no HMAC signature, dropping...")
+		return
+	}
+
+	payload := make([]byte, req.ContentLength)
+	_, err := io.ReadFull(req.Body, payload)
+	if err != nil {
+		rw.WriteHeader(http.StatusBadRequest)
+		a.logger.Debug("Failed to read alerts payload", zap.Error(err), zap.String("remote", req.RemoteAddr))
+		return
+	}
+
+	if err = verifyHMACSignature(a.secret, payload, payloadSigHeader); err != nil {
+		rw.WriteHeader(http.StatusBadRequest)
+		a.logger.Debug("Got payload with invalid HMAC signature, dropping...", zap.Error(err), zap.String("remote", req.RemoteAddr))
+		return
+	}
+
+	logs, err := payloadToLogs(time.Now(), payload)
+	if err != nil {
+		rw.WriteHeader(http.StatusBadRequest)
+		a.logger.Error("Failed to convert log payload to log record", zap.Error(err))
+		return
+	}
+
+	if err := a.consumer.ConsumeLogs(req.Context(), *logs); err != nil {
+		rw.WriteHeader(http.StatusInternalServerError)
+		a.logger.Error("Failed to consumer alert as log", zap.Error(err))
+		return
+	}
+
+	rw.WriteHeader(http.StatusOK)
+}
+
+func (a alertsReceiver) Shutdown(ctx context.Context) error {
+	a.logger.Debug("Shutting down server")
+	err := a.server.Shutdown(ctx)
+	if err != nil {
+		return err
+	}
+
+	a.logger.Debug("Waiting for shutdown to complete.")
+	a.wg.Wait()
+	return nil
+}
+
+func verifyHMACSignature(secret string, payload []byte, signatureHeader string) error {
+	b64Decoder := base64.NewDecoder(base64.StdEncoding, bytes.NewBufferString(signatureHeader))
+	payloadSig, err := io.ReadAll(b64Decoder)
+	if err != nil {
+		return err
+	}
+
+	h := hmac.New(sha1.New, []byte(secret))
+	h.Write(payload)
+	calculatedSig := h.Sum(nil)
+
+	if !hmac.Equal(calculatedSig, payloadSig) {
+		return errors.New("calculated signature does not equal header signature")
+	}
+
+	return nil
+}
+
+func payloadToLogs(now time.Time, payload []byte) (*plog.Logs, error) {
+	var alert model.Alert
+
+	err := json.Unmarshal(payload, &alert)
+	if err != nil {
+		return nil, err
+	}
+
+	logs := plog.NewLogs()
+	resourceLogs := logs.ResourceLogs().AppendEmpty()
+	logRecord := resourceLogs.ScopeLogs().AppendEmpty().LogRecords().AppendEmpty()
+
+	logRecord.SetObservedTimestamp(pcommon.NewTimestampFromTime(now))
+	logRecord.SetTimestamp(timestampFromAlert(alert))
+	logRecord.SetSeverityNumber(severityFromAlert(alert))
+	logRecord.Body().SetStringVal(string(payload))
+
+	resourceAttrs := resourceLogs.Resource().Attributes()
+	resourceAttrs.InsertString("mongodbatlas.group.id", alert.GroupID)
+	resourceAttrs.InsertString("mongodbatlas.alert.config.id", alert.AlertConfigID)
+	insertStringToMapNotNil(resourceAttrs, "mongodbatlas.cluster.name", alert.ClusterName)
+	insertStringToMapNotNil(resourceAttrs, "mongodbatlas.replica_set.name", alert.ReplicaSetName)
+
+	attrs := logRecord.Attributes()
+	// These attributes are always present
+	attrs.InsertString("event.name", alert.EventType)
```

Comment:
This aligns well with [otep 202](https://github.com/open-telemetry/oteps/pull/202), which is not yet accepted but seems like to be soon.

Another thing that's called out in that otep is that `event.name` is only meaningful within an `event.domain`, so I think we should probably add that here as well. (Can be hardcoded as `mongoatlas`, or something similar.)

Reply:
I'm a little confused on what "Scope Level" means in context of pdata, but what I got from there is that it's a log-record level attribute, so I've added it there.

Reply:
Yeah, "scope" is a newer term, replacing "instrumentation library". Additionally, there are now going to be attributes at the scope level.

So you have resource -> scope -> logrecord, each of which has its own attributes. In the context of logs, scope will commonly refer to an instance of a logger.